src/sentry/static/sentry/app/components/events/interfaces/stacktraceLinkModal.tsx

```diff
@@ -0,0 +1,207 @@
+    this.setState({
+      showModal: false,
+      sourceCodeInput: '',
+    });
+  }
+
+  createConfig = async configData => {
+    const {organization, project} = this.props;
+    const endpoint = `/organizations/${organization.slug}/integrations/${configData.integrationId}/repo-project-path-configs/`;
+    try {
+      await this.api.requestPromise(endpoint, {
+        method: 'POST',
+        data: {...configData, projectId: project.id},
+      });
+      addSuccessMessage(t('Stack trace link config created.'));
+    } catch (err) {
+      const errors = err?.responseJSON
+        ? Array.isArray(err?.responseJSON)
+          ? err?.responseJSON
+          : Object.values(err?.responseJSON)
+        : [];
+      const apiErrors = errors.length > 0 ? `: ${errors.join(', ')}` : '';
+      addErrorMessage(t('Unable to save configuration%s', apiErrors));
+    }
+    this.closeModal();
+    this.props.onClose();
+  };
+
+  onHandleChange(sourceCodeInput: string) {
+    this.setState({
+      sourceCodeInput,
+    });
+  }
+
+  handleSubmit = async () => {
+    const {sourceCodeInput} = this.state;
+    const {organization, filename, project} = this.props;
+    const endpoint = `/projects/${organization.slug}/${project.slug}/repo-path-parsing/`;
+    try {
+      const configData = await this.api.requestPromise(endpoint, {
+        method: 'POST',
+        data: {
+          sourceUrl: sourceCodeInput,
+          stackPath: filename,
+        },
+      });
+      this.createConfig(configData);
+    } catch (err) {
+      const error = err?.responseJSON?.sourceUrl;
+      if (error && error.length > 0) {
+        addErrorMessage(t(`${error[0]}`));
+      } else {
+        addErrorMessage(t('An error occurred'));
+      }
+    }
+  };
+
+  renderBody() {
+    const {showModal, sourceCodeInput} = this.state;
+    const {filename, integrations, organization} = this.props;
+    const baseUrl = `/settings/${organization.slug}/integrations`;
+
+    return (
+      <CodeMappingButtonContainer columnQuantity={2}>
+        {t('Enable source code stack trace linking by setting up a code mapping.')}
+        <Button onClick={() => this.openModal()} size="xsmall">
+          {t('Setup Code Mapping')}
+        </Button>
+        <Modal
+          show={showModal}
+          onHide={() => this.closeModal()}
+          enforceFocus={false}
+          backdrop="static"
+          animation={false}
+        >
+          <Modal.Header closeButton>{t('Code Mapping Setup')}</Modal.Header>
+          <Modal.Body>
+            <ModalContainer>
+              <div>
+                <h6>{t('Quick Setup')}</h6>
+                {tct(
+                  'Enter in your source code url that corresponsds to stack trace filename [filename]. We will create a code mapping with this information.',
+                  {
+                    filename: <code>{filename}</code>,
+                  }
+                )}
+              </div>
+              <SourceCodeInput>
```

Comment:
@MeredithAnya any reason why you didn't use a form and a form model here? That might make it easier to handle errors.

Reply:
nope no reason lol. do you want me to update?

Reply:
@MeredithAnya nah, it's probably OK with just one input